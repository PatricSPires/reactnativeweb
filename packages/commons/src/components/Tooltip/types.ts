export type TooltipProps = {
  hint: string
}
