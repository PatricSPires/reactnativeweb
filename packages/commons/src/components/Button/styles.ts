import styled, { css } from 'styled-components/native'
import { ThemeProps } from '../../styles/customThemes/types'

import { ButtonTextProps, ButtonWrapperProps } from './types'
import { getButtonAppearance, getTextAppearance } from './appearance'

export const ButtonWrapper = styled.TouchableOpacity<
  ButtonWrapperProps & ThemeProps
>`
  ${({ alignIcon, alignItems, theme }) => css`
    display: flex;
    align-items: center;
    height: 48px;
    max-width: 300px;
    min-width: 48px;
    width: 178px;
    border: none;
    padding: ${theme.spacings[2]};
    flex-direction: ${alignIcon === 'left' ? 'row-reverse' : 'row'};
    justify-content: ${alignItems === 'center' ? 'center' : 'space-between'};
    ${getButtonAppearance}
  `}
`

export const ButtonText = styled.Text<ButtonTextProps>`
  ${getTextAppearance}
`
