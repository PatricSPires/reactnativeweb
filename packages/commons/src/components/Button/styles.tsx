import React from 'react'
import styled, { css } from 'styled-components'
import { Platform, TouchableOpacity, Text } from 'react-native'
import { ButtonProps } from './types'
import { buttonAppearances, textAppearances } from './appearance'

type ButtonComponentProps = Pick<
  ButtonProps,
  | 'appearance'
  | 'marginBottom'
  | 'width'
  | 'alignIcon'
  | 'alignItems'
  | 'className'
>
type TextComponentProps = Pick<ButtonProps, 'alignText' | 'appearance'>

type ButtonComponent1Props = {
  className?: string
  children: JSX.Element
  onPress: () => void
}

const Button = styled(TouchableOpacity)`
  ${({ appearance }: ButtonComponentProps) => css`
    border: none;
    display: flex;
    flex-direction: ${({ alignIcon }: ButtonComponentProps) =>
      alignIcon === 'left' ? 'row-reverse' : 'row'};
    align-items: center;
    justify-content: ${({ alignItems }: ButtonComponentProps) =>
      alignItems === 'center' ? 'center' : 'space-between'};
    height: 48px;
    max-width: 300px;
    min-width: 48px;
    width: 178px;
    padding: 16px;

    ${Platform.OS === 'web' && `cursor: pointer;`}
    ${!!appearance && buttonAppearances[appearance]()};
  `}
`
const ButtonComponent = ({
  onPress,
  className,
  children,
  ...props
}: ButtonComponent1Props) => {
  // eslint-disable jsx-props-no-spreading
  return (
    <Button
      className={className}
      {...props}
      as={Platform.OS === 'web' ? 'button' : TouchableOpacity}
      onClick={onPress}
      onPress={onPress}
    >
      {children}
    </Button>
  )
}

ButtonComponent.defaultProps = {
  className: null,
}

export const TextComponent = styled(Text)<TextComponentProps>`
  ${({ appearance }: TextComponentProps) => css`
    ${!!appearance && textAppearances[appearance]()};
  `}
`
