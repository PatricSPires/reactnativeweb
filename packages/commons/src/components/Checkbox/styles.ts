import { Text, TouchableWithoutFeedback, View } from 'react-native'
import styled from 'styled-components/native'
import theme from '../../styles/theme'
import { CheckboxProps } from './types'

export const CheckBoxWrapper = styled(View)<CheckboxProps>`
  width: 15px;
  height: 15px;
  border: 2px solid;
  border-color: ${theme.colors.high_contrast_border};
  border-radius: 2px;
  display: flex;
  align-items: center;
  justify-content: center;

  img {
    width: 15px;
    height: 15px;
  }

  ${({ checked, disabled }: CheckboxProps) => {
    if (checked === true && disabled === false) {
      return `
        background-color: ${theme.colors.high_contrast_border};
      `
    }

    if (checked === false && disabled === true) {
      return `
        border-color: ${theme.colors.primary_subtle_border};
        background-color: transparent;
      `
    }
    return ``
  }}
`

export const CheckBoxCheck = styled(View)<CheckboxProps>`
  width: 15px;
  height: 15px;
  background-color: transparent;
  border-radius: 2px;

  ${({ checked, disabled }: CheckboxProps) => {
    if (checked === true && disabled === true) {
      return `
        background-color: ${theme.colors.high_contrast_border};
      `
    }
    return ``
  }}
`

export const CheckBoxText = styled(Text)`
  margin-left: 8px;
  color: ${(props: CheckboxProps) =>
    props.disabled
      ? theme.colors.primary_subtle_border
      : theme.colors.high_contrast_border};
`

export const Container = styled(TouchableWithoutFeedback)`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`
