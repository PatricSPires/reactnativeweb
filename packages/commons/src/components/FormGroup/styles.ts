import styled from 'styled-components'
import { View, Text } from 'react-native'
import theme from '../../styles/theme'
import { FormGroupWrapperProps } from './types'

export const FormGroupWrapper = styled(View)<FormGroupWrapperProps>`
  display: flex;
  width: ${({ width }) => width || '100%'};
  margin-bottom: ${({ marginBottom }) => theme.spacings[marginBottom || 3]};

  Tag {
    margin: 8px 0 4px 0;
  }
`

export const Label = styled(Text)`
  color: ${(props) => props.theme.colors.font_color};
  font-size: ${theme.fontSize.label_01};
  margin-bottom: ${theme.spacings[1]};
  display: flex;
  align-items: center;
`

export const HelperText = styled(Text)`
  font-size: ${theme.fontSize.label_01};
  color: ${theme.colors.medium_contrast_border};
`

export const ErrorMessage = styled(Text)`
  font-size: ${theme.fontSize.label_01};
  color: ${theme.supportColors.error};
  margin-top: ${theme.spacings[1]};
`
