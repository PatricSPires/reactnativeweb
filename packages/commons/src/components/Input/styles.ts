import styled, { css } from 'styled-components'
import {
  Text,
  TextInput as InputRN,
  View,
  TouchableWithoutFeedback,
} from 'react-native'
import { TextInputProps, IconProps, InputWrapperProps } from './types'
import theme from '../../styles/theme'

function getFocusStyles(isInvalid: boolean) {
  if (isInvalid) {
    return `
      outline: 2px solid ${theme.supportColors.error};
      border-bottom-color: transparent;
    `
  }
  return `
    outline: 2px solid ${theme.colors.primary_background};
    border-bottom-color: transparent;
  `
}

export const Wrapper = styled(View)`
  width: 100%;
`

export const InputWrapper = styled(View)<InputWrapperProps>`
  width: 100%;
  display: flex;
  flex-direction: row;
  align-items: center;
  background-color: ${(props) =>
    props.theme.colors.primary_container_background};
  border-bottom: 1px solid ${theme.colors.medium_contrast_border};

  ${({ isInvalid }) => css`
    outline: ${!!isInvalid && `2px solid ${theme.supportColors.error}`};
    border: ${!!isInvalid && 'none'};
  `}

  &:focus-within {
    ${({ isInvalid }) => getFocusStyles(isInvalid)}
  }
`

export const TextInput = styled(InputRN)<TextInputProps>`
  width: 100%;
  height: 40px;
  color: ${(props) => props.theme.colors.font_color};
  font-weight: normal;
  line-height: 18px;
  outline: transparent;
  font-size: 14px;

  order: ${({ iconPosition }) => (iconPosition === 'right' ? 0 : 1)};
  padding: ${({ iconPosition }) =>
    iconPosition === 'left' ? '0 10px 0 0' : '0 0 0 10px'};

  &:disabled {
    outline: none;
    border: none;
    color: ${(props) => props.theme.colors.font_color};
  }

  ::placeholder {
    color: ${(props) => props.theme.colors.font_color};
  }

  &:focus-within {
    color: ${(props) => props.theme.colors.font_color};
  }
`

export const HelperText = styled(Text)`
  font-size: 12px;
  color: ${theme.colors.medium_contrast_border};
`

export const ErrorMessage = styled(Text)`
  font-size: 12px;
  color: ${theme.supportColors.error};
`

export const Icon = styled(View)<IconProps>`
  order: ${({ iconPosition }) => (iconPosition === 'right' ? 1 : 0)};
  color: ${theme.colors.primary_background};
  margin: 0 10px;

  svg {
    width: 16px;
    ${({ iconColor }: IconProps) => css`
      color: ${theme.supportColors[iconColor]};
    `}
  }
`

export const PasswordIconWrapper = styled(TouchableWithoutFeedback)`
  width: 16px;
  height: 16px;
  color: ${(props) => props.theme.colors.font_color};
  margin: 0 10px;
`
