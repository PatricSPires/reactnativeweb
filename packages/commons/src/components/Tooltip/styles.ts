import styled from 'styled-components'
import { Text, View } from 'react-native'
import theme from '../../styles/theme'

export const TooltipHint = styled(Text)`
  visibility: hidden;
  width: 120px;
  background-color: ${theme.colors.high_contrast_border};
  color: ${theme.colors.primary_background};
  text-align: center;
  border-radius: 6px;
  padding: 5px 0;
  position: absolute;
  z-index: 1;
  top: -220%;
  left: 68%;
  margin-left: -60px;

  ::after {
    content: '';
    position: absolute;
    bottom: -38%;
    left: 50%;
    margin-left: -5px;
    border-width: 5px;
    border-style: solid;
    border-color: transparent transparent ${theme.colors.high_contrast_border}
      transparent;
    transform: rotate(180deg);
  }
`

export const TooltipWrapper = styled(View)`
  position: relative;
  display: flex;

  svg {
    width: 16px;
    height: 16px;
    margin-left: 10px;
  }

  :hover ${TooltipHint} {
    visibility: visible;
  }
`
