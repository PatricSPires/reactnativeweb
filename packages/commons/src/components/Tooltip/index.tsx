import React from 'react'
import { AlertCircle } from '@styled-icons/feather/AlertCircle'
import { TooltipProps } from './types'
import { TooltipWrapper, TooltipHint } from './styles'

const Tooltip = ({ hint }: TooltipProps) => (
  <TooltipWrapper>
    <AlertCircle />
    <TooltipHint>{hint}</TooltipHint>
  </TooltipWrapper>
)

export default Tooltip
