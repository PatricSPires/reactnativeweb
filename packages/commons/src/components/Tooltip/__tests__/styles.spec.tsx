import React from 'react'
import 'jest-styled-components'
import { render } from '@testing-library/react-native'
import Tooltip from '../index'
import {} from '../styles'

describe('Tooltip > Style', () => {
  it('should render the Tooltip with some styles', () => {
    const { toJSON } = render(<Tooltip hint="Hello World" />)
    expect(toJSON()).toMatchSnapshot()
  })
})
