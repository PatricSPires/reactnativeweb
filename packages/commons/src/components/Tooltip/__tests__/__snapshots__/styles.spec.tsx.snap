// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip > Style should render the Tooltip with some styles 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c3 {
  visibility: hidden;
  width: 120px;
  background-color: #161616;
  color: #ffffff;
  text-align: center;
  border-radius: 6px;
  padding: 5px 0;
  position: absolute;
  z-index: 1;
  top: -220%;
  left: 68%;
  margin-left: -60px;
}

.c3::after {
  content: '';
  position: absolute;
  bottom: -38%;
  left: 50%;
  margin-left: -5px;
  border-width: 5px;
  border-style: solid;
  border-color: transparent transparent #161616 transparent;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 svg {
  width: 16px;
  height: 16px;
  margin-left: 10px;
}

.c0:hover .c2 {
  visibility: visible;
}

<div
  className="c0 sc-gtssRu c0 css-view-1dbjc4n"
>
  <svg
    aria-hidden="true"
    className="c1"
    fill="none"
    focusable="false"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx={12}
      cy={12}
      r={10}
    />
    <line
      x1={12}
      x2={12}
      y1={8}
      y2={12}
    />
    <line
      x1={12}
      x2={12.01}
      y1={16}
      y2={16}
    />
  </svg>
  <div
    className="c2 c3 c2 c3 css-text-901oao"
    dir="auto"
  >
    Hello World
  </div>
</div>
`;
